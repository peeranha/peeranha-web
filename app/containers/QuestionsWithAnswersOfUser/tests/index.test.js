import { QuestionsWithAnswersOfUser } from '../index';

const cmp = new QuestionsWithAnswersOfUser();

cmp.props = {
  isLastFetch: false,
  questionsLoading: false,
  userId: 'userId',
  locale: 'en',
  questions: [],
  className: 'className',
  infinityOff: false,
  communities: [],
  getQuestionsDispatch: jest.fn(),
  resetStoreDispatch: jest.fn(),
};

describe('QuestionsWithAnswersOfUser', () => {
  describe('componentDidMount', () => {
    cmp.componentDidMount();
    expect(cmp.props.getQuestionsDispatch).toHaveBeenCalledWith(
      cmp.props.userId,
    );
  });

  describe('componentWillUnmount', () => {
    cmp.componentWillUnmount();
    expect(cmp.props.resetStoreDispatch).toHaveBeenCalled();
  });

  describe('fetchQuestions', () => {
    cmp.fetchQuestions();
    expect(cmp.props.getQuestionsDispatch).toHaveBeenCalledWith(
      cmp.props.userId,
    );
  });

  describe('render', () => {
    it('questionsLoading TRUE', () => {
      cmp.props.questionsLoading = true;
      expect(cmp.render()).toMatchSnapshot();
    });

    it('questions[0] TRUE', () => {
      cmp.props.questions = [
        {
          myPostRating: 100,
          title: 'title',
          myPostTime: 1550837750334,
          answers: [],
          locale: 'en',
          acceptedAnswer: false,
          communities: [],
          id: 'id',
          community_id: 123,
        },
      ];

      expect(cmp.render()).toMatchSnapshot();
    });
  });
});
