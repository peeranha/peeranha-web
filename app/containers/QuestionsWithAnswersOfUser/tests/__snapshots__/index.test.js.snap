// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsWithAnswersOfUser render questions[0] TRUE 1`] = `
<InfinityLoader
  infinityOff={false}
  isLastFetch={false}
  isLoading={true}
  loadNextPaginatedData={[Function]}
>
  <div
    className="className"
  >
    <Header />
    <QuestionsWithAnswersList
      communities={Array []}
      locale="en"
      questions={
        Array [
          Object {
            "acceptedAnswer": false,
            "answers": Array [],
            "communities": Array [],
            "community_id": 123,
            "id": "id",
            "locale": "en",
            "myPostRating": 100,
            "myPostTime": 1550837750334,
            "title": "title",
          },
        ]
      }
    />
    <LoadingIndicator />
  </div>
</InfinityLoader>
`;

exports[`QuestionsWithAnswersOfUser render questionsLoading TRUE 1`] = `
<InfinityLoader
  infinityOff={false}
  isLastFetch={false}
  isLoading={true}
  loadNextPaginatedData={[Function]}
>
  <div
    className="className"
  >
    <Header />
    <LoadingIndicator />
  </div>
</InfinityLoader>
`;
